### DISCUSSION TITLE
Using elasticsearch as jaeger storage with a fixed index

### DISCUSSION THREAD

**Discussion by stanoswald** [jaegertracing/jaeger#5009] [Q&A] [answered]
Hi,

I'm currently attempting to use elasticsearch as my storage solution for Jaeger, as my cloud service provider supports es in Serverless form, with pricing based on the number of documents in a fixed index. 

However, when I tried to configure Elasticsearch as my Jaeger storage, I encountered an error. This issue arose when the Jaeger collector attempted to create a template or index (after I disabled the `create-index-templates`). 

```
{"level":"error","ts":1702699918.853112,"caller":"config/config.go:134","msg":"Elasticsearch part of bulk request failed","map-key":"index","response":{"_index":"jaeger-span-2023-12-16","_type":"_doc","status":403,"error":{"type":"security_exception","reason":"action [indices:admin/auto_create] is unauthorized for user [elastic-bpvk5q] with roles [role_elastic-bpvk5q], this action is granted by the index privileges [auto_configure,create_index,manage,all]"}},"stacktrace":"github.com/jaegertracing/jaeger/pkg/es/config.NewClient.func2\n\tgithub.com/jaegertracing/jaeger/pkg/es/config/config.go:134\ngithub.com/olivere/elastic.(*bulkWorker).commit\n\tgithub.com/olivere/elastic@v6.2.37+incompatible/bulk_processor.go:588\ngithub.com/olivere/elastic.(*bulkWorker).work\n\tgithub.com/olivere/elastic@v6.2.37+incompatible/bulk_processor.go:501"}
```

So, I'm wondering whether there's a way to configure Jaeger to use a predefined, fixed index for storage, without the need to create new indexes or templates.

Thanks for help.

**Comment by yurishkuro** [comment, marked as answer]
It's not a supported mode. What would you do once the index gets too large?

The create templates option is different from create indices.

**Reply by stanoswald** [reply]
Still thanks for response. My original plan was to manage index size using rotation strategy which is built into the serverless es service.
